cmake/Install.cmake

```diff
@@ -9,4 +9,107 @@
 if(NOT INSTALLED_ENGINE)
     ly_get_absolute_pal_filename(pal_dir ${CMAKE_CURRENT_SOURCE_DIR}/cmake/Platform/${PAL_PLATFORM_NAME})
     include(${pal_dir}/Install_${PAL_PLATFORM_NAME_LOWERCASE}.cmake)
-endif()
\ No newline at end of file
+endif()
+
+#! ly_install_directory: specifies a directory to be copied to the install layout at install time
+#
+# \arg:DIRECTORIES directories to install
+# \arg:DESTINATION (optional) destination to install the directory to (relative to CMAKE_PREFIX_PATH)
+# \arg:EXCLUDE_PATTERNS (optional) patterns to exclude
+#
+# \notes: refer to cmake's install(DIRECTORY documentation for more information
+#
+function(ly_install_directory)
+
+    set(options)
+    set(oneValueArgs DESTINATION)
+    set(multiValueArgs DIRECTORIES EXCLUDE_PATTERNS)
+
+    cmake_parse_arguments(ly_install_directory "${options}" "${oneValueArgs}" "${multiValueArgs}" ${ARGN})
+
+    if(NOT ly_install_directory_DIRECTORIES)
+        message(FATAL_ERROR "You must provide at least a directory to install")
+    endif()
+
+    foreach(directory ${ly_install_directory_DIRECTORIES})
+
+        cmake_path(ABSOLUTE_PATH directory)
+
+        if(NOT ly_install_directory_DESTINATION)
+            # maintain the same structure relative to LY_ROOT_FOLDER
+            set(ly_install_directory_DESTINATION ${directory})
+            if(${ly_install_directory_DESTINATION} STREQUAL ".")
+                set(ly_install_directory_DESTINATION ${CMAKE_CURRENT_LIST_DIR})
+            else()
+                cmake_path(ABSOLUTE_PATH ly_install_directory_DESTINATION BASE_DIRECTORY ${CMAKE_CURRENT_LIST_DIR})
+            endif()
+            # take out the last directory since install asks for the destination of the folder, without including the fodler itself
```

Comment:
Typo: fodler

```suggestion
            # take out the last directory since install asks for the destination of the folder, without including the folder itself
```